scripts/ci/rust_checks.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
+
+"""Run various rust checks for CI."""
+
+from __future__ import annotations
+
+import os
+import re
+import subprocess
+import sys
+import time
+
+
+class Timing:
+    def __init__(self, command: str, start_time: float) -> None:
+        self.command = command
+        self.duration = time.time() - start_time
+
+
+def run_cargo(args: str) -> Timing:
+    command = f"cargo {args}"
+    print(f"Running '{command}'")
+    start = time.time()
+    result = subprocess.call(command, shell=True)
+
+    if result != 0:
+        sys.exit(result)
+
+    return Timing(command, start)
+
+
+def main() -> None:
+    timings = []
+
+    # First check with --locked to make sure Cargo.lock is up to date.
+    timings.append(run_cargo("check --locked --all-features"))
+
+    timings.append(run_cargo("fmt --all -- --check"))
+    timings.append(run_cargo("cranky --all-targets --all-features -- --deny warnings"))
+
+    # Since features are additive, check samples individually.
+    for path, dirnames, filenames in os.walk("examples/rust"):
```

Comment:
Ideally this should be done for all crates, not just the examples.

Just find all `Cargo.toml` files and return all with a `name` field (will naturally exclude the workspace `Cargo.toml`)